src/bridge.cpp

```diff
@@ -1,24 +1,133 @@
+    AABB boundaryBox(skinOutline);
+
     //To detect if we have a bridge, first calculate the intersection of the current layer with the previous layer.
     // This gives us the islands that the layer rests on.
     Polygons islands;
+
+    Polygons prevLayerOutline; // we also want the complete outline of the previous layer
+
     for(auto prevLayerPart : prevLayer->parts)
     {
+        prevLayerOutline.add(prevLayerPart.outline); // not intersected with skin
+
         if (!boundaryBox.hit(prevLayerPart.boundaryBox))
             continue;
         
-        islands.add(outline.intersection(prevLayerPart.outline));
+        islands.add(skinOutline.intersection(prevLayerPart.outline));
     }
     supportedRegions = islands;
+
+    if (supportLayer)
+    {
+        // add the regions of the skin that have support below them to supportedRegions
+        // but don't add these regions to islands because that can actually cause the code
+        // below to consider the skin a bridge when it isn't (e.g. a skin that is supported by
+        // the model on one side but the remainder of the skin is above support would look like
+        // a bridge because it would have two islands) - FIXME more work required here?
+
+        if (!supportLayer->support_roof.empty())
+        {
+            AABB support_roof_bb(supportLayer->support_roof);
+            if (boundaryBox.hit(support_roof_bb))
+            {
+                prevLayerOutline.add(supportLayer->support_roof); // not intersected with skin
+
+                Polygons supported_skin(skinOutline.intersection(supportLayer->support_roof));
+                if (!supported_skin.empty())
+                {
+                    supportedRegions.add(supported_skin);
+                }
+            }
+        }
+        else
+        {
+            for (auto support_part : supportLayer->support_infill_parts)
+            {
+                AABB support_part_bb(support_part.getInfillArea());
+                if (boundaryBox.hit(support_part_bb))
+                {
+                    prevLayerOutline.add(support_part.getInfillArea()); // not intersected with skin
+
+                    Polygons supported_skin(skinOutline.intersection(support_part.getInfillArea()));
+                    if (!supported_skin.empty())
+                    {
+                        supportedRegions.add(supported_skin);
+                    }
+                }
+            }
+        }
+    }
+
+    // if the proportion of the skin region that is supported is less than supportThreshold, it's considered a bridge and we
+    // determine the best angle for the skin lines - the current heuristic is that the skin lines should be parallel to the
+    // direction of the skin area's longest unsupported edge - if the skin has no unsupported edges, we fall through to the
+    // original code
+
+    if (supportThreshold > 0 && (supportedRegions.area() / (skinOutline.area() + 1)) < supportThreshold)
```

Comment:
The first check of supportThreshold on line 73 is testing to see if the skin region is to be considered as a bridge skin using the criteria that it is supported for less than the threshold amount. If that's true it works out if any of the skin's perimeter lines are above air and, if so, returns the orientation of the longest unsupported skin edge.

The second check of supportThreshold on line 126 is simply exiting early without going to do the original code if the skin area is supported sufficiently because as far as the new bridge code is concerned, the skin area doesn't need to be treated as a bridge.